llms/mlx_lm/sample_utils.py

```diff
@@ -22,7 +22,8 @@ def top_p_sampling(logits: mx.array, top_p: float, temperature: float) -> mx.arr
 
     # sort probs in ascending order
     sorted_indices = mx.argsort(probs, axis=-1)
-    sorted_probs = probs[..., sorted_indices]
+    # probs shape is [B,V] and after indexing it will be [1, B, V], so we squeeze it back to [B, V]
+    sorted_probs = probs[..., sorted_indices].squeeze(0)
```

Comment:
That doesn't make sense to me, where does it get the `1` from?

```python
>>> import mlx.core as mx
>>> a = mx.zeros((4, 8))
>>> a[..., mx.array([2,3])].shape
(4, 2)
```

Reply:
I think I understand now.. it's probably clearer if you do 
```python
sorted_indices = mx.argsort(probs, axis=-1)
probs[..., sorted_indices.squeeze(0)]
```

Reply:
The shape (1, B, V) is a bit surprising to me. I didn't really understand it, but it seems to be the same behavior in numpy.